index.js

```diff
@@ -1,68 +1,47 @@
+var path = require('path');
+var loaderUtils = require('loader-utils');
+var CleanCSS = require('clean-css');
+var RequestShortener = require('webpack/lib/RequestShortener');
+
+module.exports = function (content, map) {
+    var config = loaderUtils.getLoaderConfig(this, 'cssRawLoader');
 
-module.exports = function(content, map) {
-	var query = loaderUtils.parseQuery(this.query);
-    
     this.cacheable && this.cacheable();
-	this.value = content;
-    
-    if (typeof map !== 'string') {
-        map = JSON.stringify(map);
-    }
-    
-    var result = [];
-    
-    var minimize = (this && this.minimize) || false;
-    
-    if(minimize) {
-		var options = Object.create(query);
-		if(query.sourceMap && map) {
-			options.sourceMap = map;
-		}
-		var minimizeResult = new CleanCSS(options).minify(content);
-		map = minimizeResult.sourceMap;
-		content = minimizeResult.styles;
-		if(typeof map !== "string") {
-			map = JSON.stringify(map);
+    this.value = content;
+
+    if (config.minimize || this.minimize && config.minimize !== false) {
+        var options = Object.create(config);
+        if (config.sourceMap || this.sourceMap && config.sourceMap !== false) {
+            options.sourceMap = typeof map === 'string' ? map : (map ? JSON.stringify(map) : true);
+        }
+        var minimizeResult = new CleanCSS(options).minify(content);
+        map = JSON.stringify(minimizeResult.sourceMap); // Make the SourceMapGenerator a plain object
+        if (map) {
+            map = JSON.parse(map);
         }
-	} else {
-        var cssRequest = loaderUtils.getRemainingRequest(this);
-        var request = loaderUtils.getCurrentRequest(this);
+        content = minimizeResult.styles;
+    }
 
-        var sourceMap = new SourceListMap();
-        sourceMap.add(content, cssRequest, content);
-        map = sourceMap.toStringWithSourceMap({
-            file: request
-        }).map;
+    if (map && map.sources) {
+        var context = config.context || this.options.context;
+        var requestShortener = new RequestShortener(context);
+        map.sources = map.sources.map(function (source) {
+            return requestShortener.shorten(path.resolve(context, source));
+        }, this);
+        map.sourceRoot = config.sourceRoot;
+    }
 
-        if(map.sources) {
-            map.sources = map.sources.map(function(source) {
-                var p = path.relative(query.context || this.options.context, source).replace(/\\/g, "/");
-                if(p.indexOf("../") !== 0) {
-                    p = "./" + p;
-                }
-                return "/" + p;
-            }, this);
-            map.sourceRoot = "webpack://";
-        }
-        map = JSON.stringify(map);
+    if (map && config.sourceMap !== false) {
+        return 'module.exports = [[module.id, ' + JSON.stringify(content) + ', "", ' + JSON.stringify(map) + ']];';
+    } else {
+        return 'module.exports = ' + JSON.stringify(content) + ';';
```

Comment:
It seems that most loaders (at least `ExtractTextPlugin` and `style-loader`) will accept either a plain string or a list of three-long entries ([see how `style-loader` handles the result here](https://github.com/webpack/style-loader/blob/master/index.js#L25)). I'm pretty sure everything in `css-base.js` was specific to things css-loader was doing. So what this state says is if there's a map and sourcemaps aren't explicitly disabled (e.g. `css-raw-loader?-sourceMap`), return the array form with the map. Otherwise give a raw string (à la `raw-loader`).